hera_cal/tests/test_frf.py

```diff
@@ -1375,3 +1375,116 @@ def test_get_FR_buffer_from_spectra():
     assert np.isclose(actual_ratio, expected_ratio, rtol=0.25)
+
+
+def test_get_coherent_avg_design_matrix():
+    # Set up test parameters
+    baseline = np.array([14.6, 0.0, 0.0])  # 14.6m East baseline
+    lat = -0.5362  # HERA latitude in radians
+    freqs = np.linspace(100e6, 200e6, 10)  # 10 frequencies
+    old_times = np.linspace(0, 1000, 20)  # 20 integrations
+    new_times = np.linspace(250, 750, 5)  # 5 averaged integrations
+    n_samples = np.ones((freqs.size, old_times.size)) * 10
+    new_inttime = 200.0  # seconds
+    flags = np.zeros((freqs.size, old_times.size), dtype=bool)
+
+    # Test basic execution
+    design_mat = frf.get_coherent_avg_design_matrix(
+        baseline, lat, freqs, old_times, new_times, n_samples, new_inttime, flags
+    )
+
+    # Check output shape
+    assert design_mat.shape == (freqs.size, new_times.size, old_times.size)
+
+    # Check that the matrix is normalized (rows have magnitude close to 1)
+    # The sum is complex due to phase factors, so check magnitude
+    row_sums = np.sum(design_mat, axis=-1)
+    assert np.allclose(np.abs(row_sums), 1.0, rtol=0.05)
+
+    # Check that individual weights are reasonable (should be small)
+    assert np.all(np.abs(design_mat) >= 0.0)
+    assert np.all(np.abs(design_mat) <= 1.0)
+
+    # Test with flags
+    flags_copy = flags.copy()
+    flags_copy[:, :5] = True
+    design_mat_flagged = frf.get_coherent_avg_design_matrix(
+        baseline, lat, freqs, old_times, new_times, n_samples, new_inttime, flags_copy
+    )
+    # Flagged samples should have zero weight
+    assert np.allclose(design_mat_flagged[:, :, :5], 0.0)
+
+
+def test_construct_filter():
+    # Set up test parameters
+    times = np.linspace(0, 10, 100)  # 100 time samples over 10 ks
+    fc = 0.5  # center fringe-rate in mHz
+    fhw = 0.2  # half-width in mHz
+
+    # Test basic execution without weights
+    filter_mat = frf.construct_filter(times, fc, fhw)
+
+    # Check output shape (should be NxN)
+    assert filter_mat.shape == (times.size, times.size)
+
+    # Check that filter is Hermitian (for real filter operation)
+    # The filter should map data to data, so check it's a proper operator
+    assert filter_mat.dtype == np.complex128
+
+    # Test with weights
+    wgts = np.ones(times.size)
+    wgts[:10] = 0.5  # Lower weight for first 10 samples
+    filter_mat_weighted = frf.construct_filter(times, fc, fhw, wgts=wgts)
+    assert filter_mat_weighted.shape == (times.size, times.size)
+
+    # Test with different eigenvalue cutoff
+    filter_mat_loose = frf.construct_filter(times, fc, fhw, eigval_cutoff=1e-6)
+    assert filter_mat_loose.shape == (times.size, times.size)
+
+    # Test that filter preserves norm (approximately)
+    test_signal = np.ones(times.size, dtype=complex)
+    filtered = filter_mat @ test_signal
+    # For a constant signal, the filter should preserve some power
+    assert np.abs(np.sum(filtered)) > 0
```

Comment:
Filter shouldn't preserve the norm, and a DC signal should be almost completely removed with the filter settings used here. This does give me an idea for a simple, useful test though:

In both cases I'll list, use a filter centered at, e.g., 0.4 mHz with a half width of say 0.1 mHz.
* Case 1: Generate a tone with a fringe-rate of -1 mHz, assert that the filtered power is close to zero.
* Case 2: Generate a tone with a fringe-rate of 0.4 mHz, assert that the filtered power is close to the unfiltered power.